### PR TITLE
Fix bulk import messing up auto ID columns

### LINKED ISSUE #11873: [BUDI-7552] CSV import messes up auto generated Auto ID column.
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Self
  - Method: docker compose
  - Budibase Version: 2.10.10
  - App Version: 2.10.10

**Describe the bug**
When importing a CSV that has an Auto ID column into an internal Budibase table with a auto generated Auto ID column. The columns don't have awareness of what the last Auto ID is. 

For example if i imported a CSV into a new table were there hasn't been any rows created. The next manually created rows Auto ID would be 1 instead of continuing from wherever the import CSV rows left off.

**To Reproduce**
1. Create a Budibase internal db data source.
2. Create relevant columns related to the CSV below
3. Import CSV attached below
4. Add another row after CSV import
5. See issue with Auto ID columns.

**Expected behavior**
If a CSV includes an Auto ID column, it should then generate any new manually created rows + 1 of the last row. For example if the CSV imports 6 rows with related Auto Ids that add to 6, when creating the next row manually it should become 7.

**Screenshots**
Screenshot 2023-09-25 at 09 45 55
  
Screen Recording 2023-09-25 at 09 49 36

**CSV Export**
[export.csv](https://github.com/Budibase/budibase/files/12713409/export.csv)

<sub>[BUDI-7553](https://linear.app/budibase/issue/BUDI-7553/csv-import-messes-up-auto-generated-auto-id-column)</sub>

### PR DESCRIPTION
**Pull request by samwho**
## Description

When you bulk import a CSV into Budibase, and that CSV has an auto ID column in it, the `Table` is updated in memory such that the `lastID` on the auto ID field is correct, but this update is never written back to the internal database. This PR ensures that the updated table is saved so that future rows get the correct auto ID.

Addresses: 
- Fixes #11873 

## Screenshots

https://github.com/Budibase/budibase/assets/338027/e0765a34-378a-4538-89df-976f356ec4e7

## Documentation
- [x] I have reviewed the budibase documentatation to verify if this feature requires any changes. If changes or new docs are required I have written them.

### CODE REVIEW

**Comment by samwho** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/server/src/api/routes/tests/table.spec.ts [hunk 1]
```diff
@@ -182,6 +181,49 @@ describe("/tables", () => {
...
+            autocolumn: true,
+            constraints: {
+              type: "number",
+              presence: false,
+            },
+          },
+        },
+      })
+
+      let response = await request
```
Comment (mike12345567) [inline packages/server/src/api/routes/tests/table.spec.ts:203]: These tests are great - the only one small improvement we could make is to use the new test APIs to make it a bit more verbose/easier to read, these can be accessed through the `config.api` object, for example for the row creation below we could do:
```
const tableId = table._id!
let response = await config.api.row.save(tableId, {})
expect(response.body.autoId).toEqual(1)
await config.api.table.bulkImport(tableId, [{ autoId: 2 }])
response = await config.api.row.save(tableId, {})
expect(response.body.autoId).toEqual(3)
```

The only thing we would need to do here is add a `bulkImport` to the table API file, which is pretty much just your bulk API request below!

**Review by mike12345567 — APPROVED** [review]
LGTM!

File: packages/server/src/api/controllers/table/external.ts [hunk 2]
```diff
@@ -16,7 +16,9 @@ import { context, events } from "@budibase/backend-core"
 import { isRows, isSchema, parse } from "../../../utilities/schema"
 import {
   BulkImportRequest,
+  BulkImportResponse,
   Datasource,
+  FieldSchema,
```
[the remark this answers is not in the document]
Comment (samwho) [reply packages/server/src/api/controllers/table/external.ts:21]: It's not. It came in as part of one of the merges from `master` and I must have left it in by mistake.

**Pull request merged**